src/main/java/uk/ac/ebi/eva/contigalias/controller/contigalias/ContigAliasController.java

```diff
@@ -169,23 +169,58 @@ public ContigAliasController(ContigAliasHandler handler) {
         List<ChromosomeEntity> entities = handler.getChromosomesByAssemblyRefseq(refseq);
-        if (entities != null && !entities.isEmpty()) {
-            return new ResponseEntity<>(entities, HttpStatus.OK);
-        } else {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        return createAppropriateResponseEntity(entities);
+    }
+
+    @ApiOperation(value = "Get chromosomes using a combination of their own name and the Taxonomic ID's of their " +
+            "parent assemblies.",
+            notes = "Given a chromosome's name and the Taxonomic ID of the assembly that it belongs to, this endpoint" +
+                    " will return a list of chromosomes that satisfy the given parameters. Each chromosome will also " +
+                    "have it's parent assembly nested inside it. The endpoint will either return a list of " +
+                    "chromosomes or it will either return an HTTP error code 204 if no chromosomes are found or " +
+                    "return an HTTP error code 400 if invalid parameters are found.")
+    @GetMapping(value = "chromosomes/name/{name}/assembly/taxid/{taxid}")
+    public ResponseEntity<PagedModel<EntityModel<ChromosomeEntity>>> getChromosomesByChromosomeNameAndAssemblyTaxid(
+            @PathVariable @ApiParam(value = "Name of chromosome. Eg: HSCHR1_RANDOM_CTG5") String name,
+            @PathVariable @ApiParam(value = "Taxonomic ID of a group of accessions. Eg: 9606") long taxid,
+            @RequestParam(required = false) @ApiParam(value = PAGE_NUMBER_DESCRIPTION) Integer pageNumber,
+            @RequestParam(required = false) @ApiParam(value = PAGE_SIZE_DESCRIPTION) Integer pageSize) {
+        if (name == null || name.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
+        PageRequest pageRequest = createPageRequest(pageNumber, pageSize);
+        PagedModel<EntityModel<ChromosomeEntity>> pagedModel = handler.getChromosomesByChromosomeNameAndAssemblyTaxid(
+                name, taxid, pageRequest);
+        return createAppropriateResponseEntity(pagedModel);
+    }
 
+    @ApiOperation(value = "Get chromosomes using a combination of their own name and the GenBank or RefSeq accessions" +
+            " of their parent assemblies.",
+            notes = "Given a chromosome's name and the GenBank or RefSeq accession of the assembly that it belongs " +
+                    "to, this endpoint will return a list of chromosomes that satisfy the given parameters. Each " +
+                    "chromosome will also have it's parent assembly nested inside it. The endpoint will either return" +
+                    " a list of chromosomes or it will either return an HTTP error code 204 if no chromosomes are " +
+                    "found or return an HTTP error code 400 if invalid parameters are found.")
+    @GetMapping(value = "chromosomes/name/{name}/assembly/accession/{accession}")
```

Comment:
I think it would be a good idea to also change this to `chromosomes/{name}?accession={accession}`